addons/pos_sale/models/sale_order.py

```diff
@@ -30,8 +30,12 @@ def get_order_amount_unpaid(self):
         order_amount_unpaid = {}
         for sale_order in self:
             total_invoice_paid = sum(sale_order.order_line.filtered(lambda l: not l.display_type).mapped('invoice_lines').filtered(lambda l: l.parent_state != 'cancel').mapped('price_total'))
-            total_pos_paid = sum(sale_order.order_line.filtered(lambda l: not l.display_type).mapped('pos_order_line_ids.price_subtotal_incl'))
-            order_amount_unpaid[sale_order.id] = sale_order.amount_total - (total_invoice_paid + total_pos_paid)
+            #If the pos order is invoiced, the invoice line are already linked to the sale order and counted in total_invoice_paid
+            if not sale_order.pos_order_line_ids.order_id.is_invoiced:
```

Comment:
@caburj I also had to adapt this part of the code as it was failing to correctly compute the amount left to be paid.